doc/vxlan/Consistent_ecmp_for_Vxlan_tunnel.md

```diff
@@ -0,0 +1,198 @@
+# Consistent ECMP for Vxlan Tunnels
+
+# Table of Contents
+
+- [Revision](#revision)
+- [Scope](#scope)
+- [Overview](#1-overview)
+- [Schema Changes](#2-schema-changes)
+    - [Config and APP DB](#21-config-and-appdb)
+    - [STATE DB](#22-state-db)
+    - [CLI](#23-cli)
+    - [YANG model](#24-yang-model)
+- [Programming Flow](#3-programming-flow)
+- [SWSS orchagent design](#4-swss-orchagent-design)
+- [Test Plan](#5-test-plan)
+
+
+# Revision
+
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 1.0 | 11/04/2025  |     Anish Narsian  | Added Consistent hashing support  |
+
+
+# Scope
+
+This document goes over an enhancement to VXLAN tunnel endpoint ECMP to add support for consistent hashing towards a group of tunnel endpoints that are nexthops for a given tunnel route. This is an extension to the existing VNET Vxlan support as defined in the [Vxlan HLD](https://github.com/sonic-net/SONiC/blob/master/doc/vxlan/Vxlan_hld.md)
+
+
+# Abbreviations
+
+|     Abbreviation         |    Meaning      |
+|--------------------------|-----------------| 
+| NH                       | Next hop        |
+| NHG                      |  Next hop Group |
+| NHGM                      |  Next hop Group Member |
+| FG                       |  Fine Grained   |
+| ECMP                     | Equal Cost MultiPath |
+
+# 1 Overview
+The details for enabling consistent hashing for Vxlan tunnel route(VNET_ROUTE_TUNNEL) are discussed in this document.
+
+###### Use-case:
+  Vxlan tunnel routes can contain a list of endpoints(next-hops) for overlay traffic to be routed to multiple underlay endpoints(next-hops). When there are multiple endpoints, ECMP is used to select the nexthop for this traffic to be encapsulated towards and sent out. This is primarily used in scenarios where  throughput needs to be scaled beyond what a single vxlan endpoint is capable of. When these endpoints hold flow state, endpoint modifications(next-hop addition/removal), will result in most flows being rehashed and sent to a different endpoint than what they were originally going to, resulting in connection restart whenever a endpoint modification is performed. To limit connection restarts during endpoint/next hop modifications, we will enable consistent hashing for tunnel nexthops.
```

Comment:
@anish-n In a typical Leaf-Spine topology, do these tunnel endpoings all point to the same egress TOR? How is the host connected to these tunnel endpoints? A network topology diagram will quite helpful in explaining the proposed functionality.